core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -429,13 +425,44 @@ private ManifestFile filterManifestWithDeletedFiles(
     }
   }
 
-  private Evaluator strictDeleteEvaluator(PartitionSpec spec) {
-    Expression strictExpr = Projections.strict(spec).project(deleteExpression);
-    return new Evaluator(spec.partitionType(), strictExpr);
-  }
+  // an evaluator that checks whether rows in a file may/must match a given expression
+  // this class first partially evaluates the provided expression using the partition tuple
+  // and then checks the remaining part of the expression using metrics evaluators
+  private class ExpressionEvaluator {
+    private final Schema tableSchema;
+    private final ResidualEvaluator residualEvaluator;
+    private final StructLikeMap<Pair<InclusiveMetricsEvaluator, StrictMetricsEvaluator>> metricsEvaluators;
+
+    // TODO: support case sensitive flags
```

Comment:
The old implementation did not support it either.

Reply:
How difficult would it be?

Reply:
I'll do that in a follow-up PR immediately after we agree on the implementation.